test/integration/apiserver/watchcache_test.go

```diff
@@ -234,3 +238,59 @@ func BenchmarkListFromWatchCache(b *testing.B) {
 		b.Logf("Number of secrets: %d", len(secrets.Items))
 	}
 }
+
+func TestConsistentList(t *testing.T) {
+	var testcases = []struct {
+		name      string
+		fromCache bool
+	}{
+		{
+			name:      "FromCache",
+			fromCache: true,
+		},
+		{
+			name:      "FromStorage",
+			fromCache: false,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ConsistentListFromCache, tc.fromCache)()
+
+			clientSet, _, tearDownFn := framework.StartTestServer(ctx, t, framework.TestServerSetup{})
+			defer tearDownFn()
+
+			ns := framework.CreateNamespaceOrDie(clientSet, "list", t)
+			defer framework.DeleteNamespaceOrDie(clientSet, ns, t)
+
+			for i := 0; i < 10; i++ {
+				rs := newRS(ns.Name)
+				newRs, err := clientSet.AppsV1().ReplicaSets(ns.Name).Create(ctx, rs, metav1.CreateOptions{})
```

Comment:
This test is faulty - there is nothing that forces you to store Deployments and ReplicaSets in the same etcd.

We can probably fix this test by:
- creating replicaset in namespace X
- listing also replicasets, but from namespace Y

Reply:
> This test is faulty - there is nothing that forces you to store Deployments and ReplicaSets in the same etcd.

Right, but for this test we are assuming that they are served from single etcd and we want to utilize fact that those are two separate resources. 

The only way that separate resource would be aware of new revision, would be via either periodic progress notification or requesting progress notification. In this test periodic progress notification is not enabled.

Reply:
> and we want to utilize fact that those are two separate resources.

Why?